### PR TITLE
Update tests for check states

### PR DESCRIPTION
**Pull request by s-nirali**
- Tests that did not increase coverage and were redundant are removed.
- New tests reflecting the meaning of the state as explained in the
  README are added.

### CODE REVIEW

File: check/check_test.go [hunk 1]
```diff
@@ -27,10 +28,19 @@ func TestCheck_Run(t *testing.T) {
 	testCases := []TestCase{
 		{check: Check{Type: MANUAL}, Expected: WARN},
 		{check: Check{Type: "skip"}, Expected: INFO},
-		{check: Check{Type: "", Scored: false}, Expected: WARN}, // Not scored checks with no type should be marked warn
-		{check: Check{Type: "", Scored: true}, Expected: WARN},  // If there are no tests in the check, warn
-		{check: Check{Type: MANUAL, Scored: false}, Expected: WARN},
-		{check: Check{Type: "skip", Scored: false}, Expected: INFO},
+
+		{check: Check{Scored: false}, Expected: WARN}, // Not scored checks with no tests or failing tests are marked warn
```
Comment (lizrice) [inline check/check_test.go]:
```suggestion
		{check: Check{Scored: false}, Expected: WARN}, // Not scored checks with no type, or not scored failing tests are marked warn
```

File: check/check_test.go [hunk 2]
```diff
@@ -27,10 +28,19 @@ func TestCheck_Run(t *testing.T) {
...
+			check: Check{ // Not scored checks with passing tests are marked pass
+				Scored: false,
+				Audit:  ":", Commands: []*exec.Cmd{exec.Command("")},
+				Tests: &tests{TestItems: []*testItem{&testItem{}}},
+			},
+			Expected: PASS,
+		},
+
+		{check: Check{Scored: true}, Expected: WARN},                  // If there are no tests in the check, warn
+		{check: Check{Scored: true, Tests: &tests{}}, Expected: FAIL}, // If there are tests in the check, fail
```
Comment (lizrice) [inline check/check_test.go]: Is this "If there are tests **that are not passing**, mark FAIL"?

File: check/check_test.go [hunk 3]
```diff
@@ -27,10 +28,19 @@ func TestCheck_Run(t *testing.T) {
...
+
+		{check: Check{Scored: false}, Expected: WARN}, // Not scored checks with no tests or failing tests are marked warn
+		{
+			check: Check{ // Not scored checks with passing tests are marked pass
+				Scored: false,
+				Audit:  ":", Commands: []*exec.Cmd{exec.Command("")},
+				Tests: &tests{TestItems: []*testItem{&testItem{}}},
+			},
+			Expected: PASS,
+		},
```
Comment (lizrice) [inline check/check_test.go]: I guess there should also be a Scored version of this? Scored checks with passing tests are marked pass

**Comment by s-nirali** [comment]
Hello @lizrice , thank you for the review. I've updated the PR.

**Comment by lizrice** [comment]
Merged under #550

**Pull request closed without merging**